onward/app/services/PopularInTagService.scala

```diff
@@ -0,0 +1,41 @@
+package services
+
+import common.Edition
+import contentapi.ContentApiClient
+import model.RelatedContentItem
+import model.dotcomrendering.{OnwardCollectionResponse, Trail}
+import play.api.mvc.RequestHeader
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PopularInTagService(contentApiClient: ContentApiClient)(implicit
+    executionContext: ExecutionContext,
+) {
+  // `itemViewCounts` is a Map[Content.id: String, ViewCount:Int]
+  // this is generally fetched from Ophan view the MostPopularAgent, but can come from anywhere
+  // and makes this easy to test
+  def fetch(edition: Edition, tag: String, excludeTags: Seq[String], itemViewCounts: Map[String, Int])(implicit
```

Comment:
Its a bit unfortunate that `Trail.pressedContentToTrail` uses an implicit RequestHeader, but since we have it maybe its worth dropping the `edition` parameter and get it from the RequestHeader instead with `Edition(request)`

Reply:
It is - but I've left this, as we could go in and refactor that method further, but it's quite deep. Having it like this would make it easier to refactor later on.